plugins/chat/spec/services/auto_remove/handle_user_removed_from_group_spec.rb

```diff
@@ -0,0 +1,251 @@
+            group: group_2,
+            permission_type: CategoryGroup.permission_types[:full],
+          )
+          private_channel_1.add(removed_user)
+        end
+
+        context "when the user remains in one of the groups that can access a private channel" do
+          before { group_1.remove(removed_user) }
+
+          it "sets the service result as successful" do
+            expect(result).to be_a_success
+          end
+
+          it "does not remove them from that channel" do
+            expect { result }.not_to change {
+              UserChatChannelMembership.where(
+                user: [removed_user],
+                chat_channel: [private_channel_1],
+              ).count
+            }
+          end
+        end
+
+        context "when the user in remains in one of the groups but that group only has readonly access to the channel" do
+          before do
+            CategoryGroup.find_by(group: group_2, category: private_category).update!(
+              permission_type: CategoryGroup.permission_types[:readonly],
+            )
+            group_1.remove(removed_user)
+          end
+
+          it "sets the service result as successful" do
+            expect(result).to be_a_success
+          end
+
+          it "removes them from that channel" do
+            expect { result }.to change {
+              UserChatChannelMembership.where(
+                user: [removed_user],
+                chat_channel: [private_channel_1],
+              ).count
+            }.to 0
+          end
+
+          context "when the user is staff" do
+            fab!(:removed_user) { Fabricate(:admin) }
+
+            it "fails the user_not_staff policy" do
+              expect(result).to fail_a_policy(:user_not_staff)
+            end
+
+            it "does not remove them from that channel" do
+              expect { result }.not_to change {
+                UserChatChannelMembership.where(
+                  user: [removed_user],
+                  chat_channel: [private_channel_1],
+                ).count
+              }
+            end
+          end
+        end
+
+        context "when the user is no longer in any group that can access a private channel" do
+          before do
+            group_1.remove(removed_user)
+            group_2.remove(removed_user)
+          end
+
+          it "sets the service result as successful" do
+            expect(result).to be_a_success
+          end
+
+          it "removes them from that channel" do
+            expect { result }.to change {
+              UserChatChannelMembership.where(
+                user: [removed_user],
+                chat_channel: [private_channel_1],
+              ).count
+            }.to 0
+          end
+
+          context "when the user is staff" do
+            fab!(:removed_user) { Fabricate(:admin) }
+
+            it "fails the user_not_staff policy" do
+              expect(result).to fail_a_policy(:user_not_staff)
```

Comment:
We could just write `it { is_expected.to fail_a_policy(:user_not_staff) }` as it’s relatively explicit and the description is very similar to `it fails the user_not_staff policy`. Maybe that’s on purpose?

Reply:
Yes sorry I forget about the `it {is_expected}` format most of the time, will change this and other instances :)